### PR TITLE
Refined Destination Sharing

### PR DESCRIPTION
**Pull request by DnOberon**
Added the ability to manage the additional configuration needed for destinations on requests and templates. This allows users to override the defaults set by the owner of the destination at creation time. 

This is currently not hooked up and a release won't be made until the changes have been propagated to the uploaders - but the PR was getting big.

### CODE REVIEW

File: server/apps/ingest/lib/ingest/accounts.ex [hunk 1]
```diff
@@ -364,7 +364,7 @@ defmodule Ingest.Accounts do
     Ecto.Multi.new()
     |> Ecto.Multi.update(:user, User.password_changeset(user, attrs))
     |> Ecto.Multi.delete_all(:tokens, UserToken.user_and_contexts_query(user, :all))
-    |> Repo.transaction()
+    |> Repo.transaction(mode: :immediate)
```
Comment (DnOberon) [inline server/apps/ingest/lib/ingest/accounts.ex:367]: needed because sqlite's transaction mode is different

**Review by jrogerthat — COMMENTED** [review]
This looks good to me! As far as I know, those errors are all fixed.

**Review by jrogerthat — DISMISSED** [review]

File: server/apps/ingest/lib/ingest/requests.ex
Comment (jrogerthat) [inline server/apps/ingest/lib/ingest/requests.ex:1]: When I try to add a destination or metadata template, it does not reflect in the UI anymore unless I refresh the page.

**Review by jrogerthat — APPROVED** [review]
Looks good!

**Pull request merged**